Item12391: Fix for SecurityAlert CVE-2013-1666

   Item12391 applies to Foswiki 1.1.0-1.1.7. This patch covers Foswiki 1.1.3-1.1.6
   This patch updates Item12285.

~~~PATCH fee73707cdd378ca45e87a47c7198de4:92a80e84b34d0fa442e077ffc2f8a86a lib/Foswiki/Macros/MAKETEXT.pm (Foswiki-1.1.3,Foswiki-1.1.4,Foswiki-1.1.5,Foswiki-1.1.6)
--- lib/Foswiki/Macros/MAKETEXT.pm	2013-02-13 10:26:42.520780283 -0500
+++ lib/Foswiki/Macros/MAKETEXT.pm	2013-02-13 10:26:51.362682708 -0500
@@ -25,8 +25,8 @@
     $str =~ s/\]/~]/g;
 
     # restore already escaped stuff:
-    $str =~ s/~~\[/~[/g;
-    $str =~ s/~~\]/~]/g;
+    $str =~ s/~~+\[/~[/g;
+    $str =~ s/~~+\]/~]/g;
 
     $max         = 0;
     $min         = 1;

